Userland/Libraries/LibGfx/JPEGLoader.cpp

```diff
@@ -363,6 +363,31 @@ static ErrorOr<void> build_macroblocks(JPEGLoadingContext& context, Vector<Macro
     return {};
 }
 
+static bool is_dct_based(StartOfFrame::FrameType frame_type)
+{
+    return frame_type == StartOfFrame::FrameType::Baseline_DCT
+        || frame_type == StartOfFrame::FrameType::Extended_Sequential_DCT
+        || frame_type == StartOfFrame::FrameType::Progressive_DCT
+        || frame_type == StartOfFrame::FrameType::Differential_Sequential_DCT
+        || frame_type == StartOfFrame::FrameType::Differential_Progressive_DCT
+        || frame_type == StartOfFrame::FrameType::Progressive_DCT_Arithmetic
+        || frame_type == StartOfFrame::FrameType::Differential_Sequential_DCT_Arithmetic
+        || frame_type == StartOfFrame::FrameType::Differential_Progressive_DCT_Arithmetic;
+}
+
+static void reset_decoder(JPEGLoadingContext& context)
+{
+    // E.2.4 Control procedure for decoding a restart interval
+    if (is_dct_based(context.frame.type)) {
+        context.previous_dc_values[0] = 0;
+        context.previous_dc_values[1] = 0;
+        context.previous_dc_values[2] = 0;
+        return;
+    }
+
+    VERIFY_NOT_REACHED();
```

Comment:
this is true because we reject lossless frametypes further up, right?

Reply:
Absolutely, everything that isn't a SOF0 is not even read. And reaching EOI early returns an error.
So if you hit the VERIFY_NOT_REACHED(); it's because you touched something and needs to update it.
Maybe I should change it to a TODO()?

Reply:
Meh, fine as-is imho. But also doesn't matter too much either way.